torch/csrc/jit/codegen/cuda/inline_propagator.cpp

```diff
@@ -126,7 +126,23 @@ size_t MaxPosCalculator::getMaxProducerPosFromConsumer(
       BestEffortReplay::replayCasP(consumer, producer, -1, pairwise_root_map);
   auto p2c_replay_map = replay_CasP.getReplay();
 
-  for (size_t producer_pos = 0; producer_pos < producer->nDims();
+  // Get the maximum position that the producer can be replayed as the consumer
+  // without mismatch. This is the maximum position that we can inline into
+  // because if the producer has a inline position larger than that, then the
+  // max producer position of the consumer will become invalid and expression
+  // sort will not be possible.
+  size_t max_producer_pos = 0;
+  for (int64_t consumer_pos = consumer->nDims(); consumer_pos >= 0;
+       consumer_pos--) {
+    auto producer_pos = TransformReplay::getMatchedLeafPosWithoutReplayPasC(
+        producer, consumer, consumer_pos);
+    if (producer_pos >= 0) {
+      max_producer_pos = producer_pos;
+      break;
+    }
+  }
```

Comment:
Note to myself: clean this up when https://github.com/csarofeen/pytorch/pull/1870 is merged

Reply:
rebased and cleaned up